concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/ProtocolLevelTokens.hs

```diff
@@ -324,3 +326,36 @@ createToken config plts = do
     (tokIndex, newTable) <- LFMBTree.append plt (_pltTable plts)
     let newMap = Map.insert (_pltTokenId config) tokIndex (_pltMap plts)
     return (tokIndex, ProtocolLevelTokens{_pltTable = newTable, _pltMap = newMap})
+
+-- | The table of PLTs of a given account. The table is indexed by the token
+--  index into the global token table.
+newtype TokenAccountStateTable = TokenAccountStateTable
+    { tokenAccountStateTable :: [(TokenIndex, TokenAccountState)]
+    }
+    deriving newtype (Eq, Ord, Show, Serialize)
```

Comment:
I'd suggest using a `Map` for this, and storing the `TokenAccountState` under a reference

Reply:
Should this be a `BufferedRef` or a `HashedBufferedRef`?

Reply:
Also, how do we decide when to use these references?

Reply:
I think maybe `HashedBufferedRef`, assuming the hash is being used. I am not sure that we have a good rule of thumb for what reference type to use where. I think we should come back and review our choices at some point.

Reply:
This is changed accordingly.